cmd/producer/main/main_test.go

```diff
@@ -83,264 +72,3 @@ func TestSuite(t *testing.T) {
-		}
-	}()
-	ln, err := net.Listen("tcp", "localhost:9001")
-	if err != nil {
-		t.Errorf("failed to start server:\n%s", err.Error())
-	}
-	byteChan := make(chan []byte)
-	debug := false
-
-	go RunServer(ln, byteChan, debug)
-	testMode := true
-	prodInterval := "2000ms"
-	memStats := false
-	fl := Flags{
-		debug:       &debug,
-		interval:    &prodInterval,
-		test:        &testMode,
-		memoryStats: &memStats,
-	}
-	senderPrivateKey, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	recipientPrivateKey, _ := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	recipientPublicKeyHash := block.HashSHA256(keys.EncodePublicKey(&recipientPrivateKey.PublicKey))
-	contract, _ := accounts.MakeContract(1, senderPrivateKey, recipientPublicKeyHash, 1000, 1)
-	contract.SignContract(senderPrivateKey)
-	serializedContract, _ := contract.Serialize()
-
-	var contractMessage []byte
-	contractMessage = append(contractMessage, producer.SecretBytes...)
-	contractMessage = append(contractMessage, 1)
-	contractMessage = append(contractMessage, serializedContract...)
-
-	conn, err := net.Dial("tcp", "localhost:9001")
-	if err != nil {
-		t.Errorf("failed to connect to server:\n%s", err.Error())
-	}
-	_, err = conn.Write(contractMessage)
-	if err != nil {
-		t.Errorf("failed to send message")
-	}
-	ProduceBlocks(byteChan, fl, true)
-
-	youngestBlock, err := blockchain.GetYoungestBlock(ledger, metadataTable)
-	if err != nil {
-		t.Errorf("failed to get youngest block:\n%s", err.Error())
-	}
-	data := youngestBlock.Data[0]
-	var compContract accounts.Contract
-	if err := compContract.Deserialize(data); err != nil {
-		t.Errorf("failed to deserialize data:\n%s", err.Error())
-	}
-	if !bytes.Equal(serializedContract, data) {
-		t.Errorf("data does not match:\n%s != %s", string(serializedContract), string(data))
-	}
-}
-
-func TestResponseToAccountInfoRequest(t *testing.T) {
-	if err := client.SetupWallet(); err != nil {
-		t.Errorf("failed to setup wallet:\n%s", err.Error())
-	}
-	defer func() {
-		if err := os.Remove("aurum_wallet.json"); err != nil {
-			t.Errorf("failed to remove aurum_wallet.json:\n%s", err.Error())
-		}
-	}()
-	dbName := "accountBalanceTable.tab"
-	dbc, _ := sql.Open("sqlite3", dbName)
-	defer func() {
-		err := dbc.Close()
-		if err != nil {
-			t.Errorf("Failed to remove database: %s", err)
-		}
-		err = os.Remove(dbName)
-		if err != nil {
-			t.Errorf("Failed to remove database: %s", err)
-		}
-	}()
-	statement, _ := dbc.Prepare("CREATE TABLE IF NOT EXISTS account_balances (public_key_hash TEXT, balance INTEGER, nonce INTEGER)")
-	statement.Exec()
-	walletAddress, err := client.GetWalletAddress()
-	// t.Logf("Wallet address: %v", walletAddress)
-	if err != nil {
-		t.Errorf("failed to retrieve wallet address:\n%s", err.Error())
-	}
-	ln, err := net.Listen("tcp", "localhost:10000")
-	if err != nil {
-		t.Errorf("failed to start server:\n%s", err.Error())
-	}
-	byteChan := make(chan []byte)
-	debug := false
-
-	go RunServer(ln, byteChan, debug)
```

Comment:
I believe this is a classic case of two threads racing each other to the end. This is just all the more reason that we should be using docker tests for communication tests. I'm curious, what happens if you put a `time.Sleep()` (https://golang.org/pkg/time/#Sleep) right after `RunServer` ? Set it to like 3 seconds.

Reply:
The test passes on my machine, but I'm not sure about Travis.

Reply:
What happens if you run test at project root?

Reply:
![image](https://user-images.githubusercontent.com/42977312/59728961-da80af80-9201-11e9-8116-1b6c5ac05a28.png)
All the tests passed.